src/LogDensityProblems.jl

```diff
@@ -124,6 +127,31 @@ The first argument (the log density) can be shifted by a constant, see the note
 """
 function logdensity_and_gradient end
 
+"""
+    logdensity_gradient_and_hessian(ℓ, x)
+
+Evaluate the log density `ℓ`, its gradient, and Hessian at `x`, which
+has length compatible with its [`dimension`](@ref).
+
+Return three values:
+
+- the log density as real number, which equivalent to `logdensity(ℓ, x)`
+
+- *if* the log density is finite, the gradient, an `::AbstractVector` of real numbers,
+   otherwise this value is arbitrary and should be ignored.
+
+- *if* the log density is finite, the Hessian, an `::AbstractMatrix` of real numbers,
```

Comment:
Should the interface include a note about non-finite gradient (elements) as well?

Reply:
My intention for the API is the implicit assumption that if the log density is finite, so are all derivatives, but this is not explicitly documented.

The motivation for the above is that the caller only needs to check the log density to make sure things make sense. I will open a separate issue for this though as it goes beyond this PR.

Reply:
Cf #102.